src/mqt/qecc/circuit_synthesis/encoding.py

```
@@ -11,73 +11,506 @@
     import numpy.typing as npt
 
-    from ..codes.css_code import CSSCode
+    from ..codes import StabilizerCode
 
 
 logger = logging.getLogger(__name__)
 
 
-def heuristic_encoding_circuit(code: CSSCode, optimize_depth: bool = True, balance_checks: bool = False) -> CNOTCircuit:
-    """Synthesize an encoding circuit for the given CSS code using a heuristic greedy search.
+def depth_optimal_encoding_circuit_non_css(
```

Ruff (0.15.12)

[warning] Too many branches (51 > 12)

(PLR0912)